dalite_xblock/public/js/dalite_xblock.js

```diff
@@ -4,8 +4,97 @@ function DaliteXBlock(runtime, element) {
     // hack to make LTI Consumer css applied to Dalite XBlock
     $(element).addClass("xblock-student_view-lti_consumer xblock-student_view");
     $(element).children(".xblock-dalite").addClass("lti_consumer");
-    
+
+    $(function ($) {
```

Comment:
Hmm, the lean code already exists in [platform](https://github.com/edx/edx-platform/blob/76b8e2e89761c30ac7aa2fa5e8aca4dc35d9002b/common/static/js/vendor/jquery.leanModal.js) as well as [xblock-lti-consumer](https://github.com/edx/xblock-lti-consumer/blob/24ef0b15/lti_consumer/static/js/xblock_lti_consumer.js#L2), so this is a third copy of the same code, which is a bit unfortunate. I do appreciate the argument that doing it this way means we own the code though. We have had our XBlocks broken by upstream JS changes fairly often in the past :-/

For what it's worth, you could use the upstream code if you wanted by adding a couple extra event handlers. Notice that there are [only two possible ways](https://github.com/edx/xblock-lti-consumer/blob/24ef0b15/lti_consumer/static/js/xblock_lti_consumer.js#L22-L27) to trigger the `close_modal` method in the upstream XBlock's JS code. We just need to catch those events, and then we can attach the refreshXBlock handler to them.

```javascript
var modal_selector = $element.data("target");
var overlay_id = (modal_selector + '_lean-overlay').replace('#', '');
var onClose = function () { runtime.refreshXBlock($element.data("data-xblock-id")); };
$("#" + overlay_id).on("click", onClose);
$(modal_selector).on("click", ".close-modal", onClose);
```